dotnetapp-buildcontainer/build.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+# Capture the current build and app image, if any, so that we can clean them up when we are done.
+currentBuildImage=$(docker images build-image -q)
+currentAppImage=$(docker images dockerapp -q)
+
+docker build -t build-image -f Dockerfile.build .
+
+# Create a container from our build image so that we can copy the contents out.
+docker create --name build-container build-image
+# Copy the contents of the /out directory out so that we can build our app image.
+docker cp build-container:/out ./out
+# Build the application image.
+docker build -t dockerapp -f ./out/Dockerfile ./out
+
+# Cleanup
+docker rm build-container
+
+rm -r ./out
+
+if [ -n "$currentBuildImage" ]; then
```

Comment:
Can you explain what the goal of this cleanup logic is?  On a clean machine, run the script twice back to back.  It appears that on the second run, the cleanup logic ends up deleting the images that were just built.  Additionally the script errors out if there is a container that is referencing the app image.